classes/Upgrader.php

```diff
@@ -121,17 +126,30 @@ public function getDestinationVersion(): ?string
     }
 
     /**
+     * @throws DistributionApiException
      * @throws UpgradeException
      */
     public function getLatestModuleVersion(): string
     {
-        $channelFile = $this->fileLoader->getXmlChannel();
+        $autoupgradeReleases = $this->distributionApiService->getAutoupgradeCompatibilities();
+
+        if (empty($autoupgradeReleases)) {
+            throw new UpgradeException($this->translator->trans('Unable to retrieve autoupgrade releases.'));
+        }
+
+        $destinationVersion = $this->getDestinationVersion();
+
+        $eligibleAutoupgradeReleases = array_filter($autoupgradeReleases, function ($autoupgradeRelease) use ($destinationVersion) {
+            return $autoupgradeRelease->getPrestashopMinVersion() <= $destinationVersion && $autoupgradeRelease->getPrestashopMaxVersion() >= $destinationVersion;
+        });
+
+        $autoupgradeRelease = reset($eligibleAutoupgradeReleases);
 
-        if (empty($channelFile)) {
-            throw new UpgradeException($this->translator->trans('Unable to retrieve channel.xml.'));
+        if ($autoupgradeRelease === false) {
+            throw new UpgradeException($this->translator->trans('Unable to retrieve autoupgrade recommended release.'));
```

Comment:
```suggestion
            throw new UpgradeException($this->translator->trans('Unable to retrieve the recommended release of Update Assistant.'));
```

Reply:
This needs to be checked, but I'm wondering if upgrades to PrestaShop will trigger an HTTP 500